--- a/src/cpp/CMakeLists.txt	2026-02-01 18:38:07.620713248 +0100
+++ b/src/cpp/CMakeLists.txt	2026-02-01 21:42:11.878893981 +0100
@@ -116,7 +116,6 @@
 	set_target_properties(web-ifc-mt PROPERTIES LINK_FLAGS "${DEBUG_FLAG} -pthread -s PTHREAD_POOL_SIZE=navigator.hardwareConcurrency --bind -flto --define-macro=REAL_T_IS_DOUBLE -sSTACK_SIZE=5MB -s ALLOW_MEMORY_GROWTH=1 -s MAXIMUM_MEMORY=4GB -s EXPORT_NAME=WebIFCWasm -s MODULARIZE=1 -s ENVIRONMENT=web,worker")
 endif()
 
-if(NOT EMSCRIPTEN)
 	include(GNUInstallDirs)
 	include(CMakePackageConfigHelpers)
 
@@ -132,6 +131,7 @@
 		$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/web-ifc>
 	)
 
+if(NOT EMSCRIPTEN)
 	# build parameters for web-ifc-test
 	add_executable(web-ifc-test ${web-ifc-source} "./test/encoding_test.cpp" "./test/main.cpp" "./test/io_helpers.cpp")
 	param_setter(web-ifc-test)
@@ -150,6 +150,7 @@
 		target_compile_options(web-ifc PUBLIC "-DCSG_DEBUG_OUTPUT")
 		target_compile_options(web-ifc PUBLIC "-DDEBUG_DUMP_SVG")
 	endif()
+endif()
 
 	# Installation rules
 	# Install the static library
@@ -210,4 +211,3 @@
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/web-ifc-config.cmake
 		DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/web-ifc
 	)
-endif()
